php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -163,13 +163,14 @@ class NDB_BVL_Instrument extends NDB_Page
      * @return object the new object of $instrument type
      * @access public
      */
-    static function factory($instrument, $commentID, $page, $guarantee_exists = true)
-    {
+    static function factory(
+        $instrument,
+        $commentID = '',
+        $page = '',
+        $guarantee_exists = true
+    ) {
```

Comment:
While you're at it, should this parameters be typed?

Reply:
valid point but not in a bugfix, risk of breaking silent failures. the additions above only get applied if entry is already null, they dont impose anything